tests/integration/security/file_mount_citadel_flow/file_mount_flow_test.go

```diff
@@ -0,0 +1,81 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package filemountcitadelflow
+
+import (
+	"testing"
+	"time"
+
+	"istio.io/istio/pkg/test/framework/components/environment/kube"
+
+	"istio.io/istio/tests/integration/security/util"
+
+	"istio.io/istio/pkg/test/echo/common/scheme"
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/echo"
+	"istio.io/istio/pkg/test/framework/components/echo/echoboot"
+	"istio.io/istio/pkg/test/framework/components/environment"
+	"istio.io/istio/pkg/test/framework/components/istio"
+	"istio.io/istio/pkg/test/framework/components/namespace"
+	"istio.io/istio/pkg/test/util/retry"
+	"istio.io/istio/tests/integration/security/util/connection"
+)
+
+func TestFileMountCitadelCaFlow(t *testing.T) {
+	framework.NewTest(t).
+		RequiresEnvironment(environment.Kube).
+		Run(func(ctx framework.TestContext) {
+
+			istioCfg := istio.DefaultConfigOrFail(t, ctx)
+
+			namespace.ClaimOrFail(t, ctx, istioCfg.SystemNamespace)
+			ns := namespace.NewOrFail(t, ctx, "file-mount-citadel-flow", true)
+
+			var a, b echo.Instance
+			echoboot.NewBuilderOrFail(t, ctx).
+				With(&a, util.EchoConfig("a", ns, false, nil, g, p)).
+				With(&b, util.EchoConfig("b", ns, false, nil, g, p)).
+				BuildOrFail(t)
+
+			checkers := []connection.Checker{
+				{
+					From: a,
+					Options: echo.CallOptions{
+						Target:   b,
+						PortName: "http",
+						Scheme:   scheme.HTTP,
+					},
+					ExpectSuccess: true,
+				},
+			}
+
+			// Sleep 10 seconds for the policy to take effect.
+			time.Sleep(10 * time.Second)
```

Comment:
Is this the recommended way to write tests ?

Reply:
I find this being used in multiple places. For example, [TestReachability](https://github.com/istio/istio/blob/153851a63e275effbcecd2f98b0972d47cdb22c8/tests/integration/security/reachability/reachability_test.go#L197).

I referred to test [TestSdsCitadelCaFlow](https://github.com/istio/istio/blob/153851a63e275effbcecd2f98b0972d47cdb22c8/tests/integration/security/sds_citadel_flow/sds_citadel_flow_test.go#L63), which also use this way.

Reply:
@JimmyCYJ it it possible to make this event driven?  E.g., poll on something until you see that the policy has been applied, but wait no more than 10 seconds?

I agree lots of other tests have sleeps like this, but we should try to avoid spreading this bad practice to new tests